src/broadcast/participant.rs

```diff
@@ -54,6 +54,10 @@ pub enum Status {
 
 #[derive(Debug)]
 pub(crate) struct BroadcastParticipant {
+    /// The current session identifier
+    sid: Identifier,
+    /// The current protocol input
+    input: (),
```

Comment:
suggestion: since this is always going to be `()`, we don't necessarily have to keep it as a field here. If we can return `&()` idrectly from the `input(&self)` function, I'd say let's remove this field.